pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/SortOperator.java

```diff
@@ -137,8 +153,15 @@ private void consumeInputBlocks() {
         }
 
         List<Object[]> container = block.getContainer();
-        for (Object[] row : container) {
-          SelectionOperatorUtils.addToPriorityQueue(row, _rows, _numRowsToKeep);
+        if (_priorityQueue == null) {
+          // TODO: when push-down properly, we shouldn't get more than _numRowsToKeep
+          if (_rows.size() <= _numRowsToKeep) {
+            _rows.addAll(container);
```

Comment:
should we add only `_numRowsToKeep  - _rows.size()` number of rows vs all of container ?

Reply:
yeah sounds good. i put the trimming at return block instead of insert but both are correct